addons/sourcemod/scripting/surftimer/buttonpress.sp

```diff
@@ -89,36 +92,111 @@ public void CL_OnStartTimerPress(int client)
 				g_bMissedBonusBest[client] = false;
 
 			}
-		}
 
-		if (!g_bPracticeMode[client] && !IsFakeClient(client)) {
-			char szSpeed[128];
-			char preMessage[128];
-			Format(szSpeed, sizeof(szSpeed), "%i", RoundToNearest(g_fLastSpeed[client]));
-			Format(preMessage, sizeof(preMessage), "%t", "StartPrestrafe", g_szChatPrefix, szSpeed);
-			if (g_iPrespeedText[client])
-				CPrintToChat(client, preMessage);
-		
-			for (int i = 1; i <= MaxClients; i++) {
-				if (!IsClientInGame(i))
-					continue;
+			//PrestrafeMsgStart
+			if (!g_bPracticeMode[client])
+			{
+				g_iStartVelsNew[client][zgrp][0] = AllSpeed[0];// XY
+				g_iStartVelsNew[client][zgrp][1] = AllSpeed[1];// XYZ
+				g_iStartVelsNew[client][zgrp][2] = AllSpeed[2];// Z
+
+				int idiff;
+				int guess = g_SpeedMode[client];
```

Comment:
Why "guess" and not like "iSpeedMode"?

Reply:
because we are guessing what speedmode client is using, no real reason tbh

Reply:
xD but its not really a guess if we know what they have selected "g_SpeedMode[client]"